crates/sui-core/src/authority_server.rs

```diff
@@ -1235,11 +1235,10 @@ impl ValidatorService {
         let mut cur_status = match first_status {
             NotifyReadConsensusTxStatusResult::Status(status) => match status {
                 ConsensusTxStatus::Rejected => {
-                    let response = WaitForEffectsResponse::Rejected {
-                        // TODO(fastpath): cache reject reason during voting and return it here.
-                        error: SuiError::Unknown("TODO: use cached reject reason".to_string()),
-                    };
-                    return Ok(response);
+                    let error = epoch_store
+                        .get_rejection_vote_reason(consensus_position)
+                        .unwrap_or(SuiError::Unknown("Unknown reject reason".to_string()));
```

Comment:
`SuiError::Unknown` was a placeholder and maybe a bad choice. How about allow `WaitForEffectsResponse::Rejected` to have a `None` error, or use `SuiError::TransactionNotFound`, or add a new error?